pennylane/templates/subroutines/single_excitation_op.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``SingleExcitationOp`` template.
+"""
+from pennylane import numpy as np
+
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.ops import CNOT, RX, RZ, Hadamard
+from pennylane.templates.decorator import template
+from pennylane.templates.utils import (
+    check_no_variable,
+    check_shape,
+    check_type,
+    check_wires,
+    get_shape,
+)
+
+
+@template
+def SingleExcitationUnitary(weight, wires=None):
+    r"""Circuit to exponentiate the tensor product of Pauli matrices representing the
+    fermionic single-excitation operator entering the Unitary Coupled-Cluster Singles
+    and Doubles (UCCSD) ansatz. UCCSD is one of the VQE ansatz commonly used to run quantum
```

Comment:
```suggestion
    and Doubles (UCCSD) ansatz. UCCSD is a VQE ansatz commonly used to run quantum
```

Just to avoid the unknown plural ;)